web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/svp/BekreftSvangerskapspengerOppdaterer.java

```diff
@@ -93,7 +93,7 @@ public OppdateringResultat oppdater(BekreftSvangerskapspengerDto dto, Aksjonspun
         verifiserUnikeDatoer(dto);
         var behandling = behandlingRepository.hentBehandling(param.getBehandlingId());
         var termindatoEndret = oppdaterFamiliehendelse(dto, behandling);
-        var tilretteleggingEndret = oppdaterTilrettelegging(dto, behandling);
+        var tilretteleggingEndret = oppdaterTilretteleggingOgLagHistorikkHvisEndret(dto, behandling);
```

Comment:
Ville være enda bedre om man bare hadde utledet om det er endring først og så kjøre videre med foretningslogikken.

Reply:
Er mange if (erEndret) i denne koden - lurte på om man ikke kunne kjøre det litt mer ala boolean erEndret = utledEndring(...) 

if (erEndret) {
   gjørDetSomSkalGjøresVedEnEndring(...)
} else {
   gjørKunDetSomSkalGjøresUtenEndringEllerIngenting(...)
}

Reply:
enig at det fortsatt ikke er optimalt, men en del av forretningslogikken her er å avklare om noe er endret. Og for å finne ut om det er en endring må man løpe gjennom hvert arbeidsforhold med dens tilrettelegginger osv..og mappe om til tilrettelegginger som kan sammenlignes så jeg ser ikke helt hvordan jeg skal få til det enklere uten større omskriving.. Jeg tar gjerne i mot forslag. Eller kanskje navngivningen ble misvisende?

Reply:
Mulig jeg ser en annen løsning - prøver ut

Reply:
har lagt inn en nytt forslag i commit 2.